mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/AnnotationMapper.java

```diff
@@ -24,58 +24,64 @@
 
 import org.apache.james.mailbox.model.MailboxAnnotation;
 import org.apache.james.mailbox.model.MailboxAnnotationKey;
+import org.apache.james.mailbox.model.MailboxId;
 import org.apache.james.mailbox.store.transaction.Mapper;
 
 public interface AnnotationMapper extends Mapper {
-
     /**
-     * Return all <code>MailboxAnnotation</code>.
+     * Return all <code>MailboxAnnotation</code> of specific mailbox.
      * The result is not ordered and should not contain duplicate by key
-     * 
+     *
+     * @param mailboxId the selected mailbox
      * @return List<MailboxAnnotation>
      */
-    List<MailboxAnnotation> getAllAnnotations();
+    List<MailboxAnnotation> getAllAnnotations(MailboxId mailboxId);
```

Comment:
All your commits should be squashed in one.
All the code has to compile on each commit.